p2p/simulations/http_test.go

```diff
@@ -517,3 +624,129 @@ func TestHTTPSnapshot(t *testing.T) {
 		x.connEvent(nodes[0].ID, nodes[1].ID, true),
 	)
 }
+
+//Following test tests message filtering
+//It sets up a filter with multiple protocols
+//It expects the messages to pass
+func TestMsgFilterPassMultiple(t *testing.T) {
+	// start the server
+	network, s := testHTTPServer(t)
+	defer s.Close()
+
+	client := NewClient(s.URL)
+	events := make(chan *Event, 100)
+	//TODO: add more exhaustive unit tests for the filterstr
+	filterstr := "prb:0;test:0"
+	opts := SubscribeOpts{
+		Current: false,
+		Filter:  filterstr,
+	}
+	sub, err := client.SubscribeNetwork(events, opts)
+	if err != nil {
+		t.Fatalf("error subscribing to network events: %s", err)
+	}
+	defer sub.Unsubscribe()
+	// check we got all the events
+
+	nodeCount := 30
+	startNetwork(network, client, t, nodeCount)
+	x := &expectEvents{t, events, sub}
+
+	filters := make(map[MsgFilter]struct{})
+	filters[MsgFilter{Proto: "prb", Code: 0}] = struct{}{}
+	filters[MsgFilter{Proto: "dum", Code: 1}] = struct{}{}
+	filters[MsgFilter{Proto: "test", Code: 0}] = struct{}{}
+	x.expectMsg(filters, 100)
```

Comment:
Why are we only expecting 100 messages here? Won't there be 30 (nodeCount) * 2 (connsPerNode) * 3 (msgsMatchingFilter) = 180?